### PR TITLE
iox-#1410 ice-env.sh supports parallel docker containers

### LINKED ISSUE #1410: Iceoryx Environment should have a separate ice_env container for every OS_VERSION
## Brief feature description

At the moment the docker container remains when one reboots the computer which leads to an error when the docker should be started again.
We could simply reused it.

When one hops between multiple OS_VERSIONS the ice_env container has to be deleted first and a new one has to be created and setup which takes time. It would be better when the container would be named ice_env_OS_VERSION so that it can be reused when needed.

The ice_env can get a list command to list all ice_env container and also a drop command to delete a container when it is no longer needed.

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [ x Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1410

### CODE REVIEW

**Review by elBoberido — DISMISSED** [review]
If you like, you could change the `WHITE_COLOR` to e.g. just bold, just in case someone has a color scheme with a light background

**Review by elBoberido — APPROVED** [review]

File: .lycheeignore [hunk 1]
```diff
@@ -1,6 +1,3 @@
-https://github.com/eclipse-iceoryx/iceoryx/issues/.*
-https://github.com/eclipse-iceoryx/iceoryx/pull/.*
```
Comment (elBoberido) [inline .lycheeignore:2]: It seems this results in many link checker network errors. Maybe it's better to revert

Comment (elfenpiff) [reply .lycheeignore:2]: No need for that anymore with a github token and a max_concurrency of 1 it runs smoothly.

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks for the fix!

**Pull request merged**